server/src/main/java/org/opensearch/index/translog/WriteOnlyTranslogManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog;
+
+import org.apache.lucene.store.AlreadyClosedException;
+import org.opensearch.common.util.concurrent.ReleasableLock;
+import org.opensearch.index.engine.EngineConfig;
+import org.opensearch.index.seqno.LocalCheckpointTracker;
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.index.translog.listener.TranslogEventListener;
+
+import java.io.IOException;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/***
+ * The implementation of {@link TranslogManager} that only orchestrates writes to the underlying {@link Translog}
+ *
+ * @opensearch.internal
+ */
+public class WriteOnlyTranslogManager extends InternalTranslogManager {
```

Comment:
I am curios why we need the `WriteOnlyTranslogManager` implementation? If write / read are supposed to be logically separated, we should better design that as Translog-related interfaces rather than throwing `UnsupportedOperationException`

Reply:
That's kind of my point (sorry for not being clear): `TranslogManager` combines translog (write) and recovery (read) at the same time. Functionally wise, they could be separated into TranslogManager` and `TranslogRecoveryManager` (or along these lines, `TranslogRecoverySupport` fe) and this is what you are trying to solve here with `UnsupportedOperationException`. But splitting interfaces, you would clearly separate write and read paths (hope it makes sense).

Reply:
:+1: seems like you have put enough thoughts on it, may be later the separation would become apparent